fpga_builder/builder.py

```diff
@@ -331,8 +325,9 @@ def line_handler(line):
         pin_txt = get_changeset_numbers()
         pin_file = output_dir / "pin.txt"
         pin_file.write_text(pin_txt)
-        filename = f"{get_app_name()}-{deployer.get_current_branch()}-{deployer.get_current_commit_hash()}.tar.xz"
-        tar_target = output_dir / filename
+        branch = deployer.get_current_branch() if not build_args.branch else build_args.branch
```

Comment:
I think this is supposed to be done with `build_args.branch is None`.  Not 100% sure the difference.  Technically what you have would evaluate to true if the branch variable was `False`.  Which sounds like a stupid thing for it to be, but could happen.